torch/distributed/rpc/backend_registry.py

```diff
@@ -76,7 +76,7 @@ def process_group_init_backend_handler(
                 )
             )
         # TODO: add try-except and destroy _agent in all processes if any fails.
-        return dist.ProcessGroupAgent(self_name, group, num_send_recv_threads)
+        return dist.ProcessGroupAgent(self_name, group, num_send_recv_threads, kwargs["rpc_timeout"])
```

Comment:
Can we put this into the argument list instead, I think this arg is universal to all RPC backends. So shouldn't unpack it from kwargs.

As regard to `num_send_recv_threads`, it's an arg specific to `ProcessGroupAgent`, we could later remove it from the explicit argument list, and allow users to pass it (like any other extra args) through `kwargs`.

Reply:
This is done in https://github.com/pytorch/pytorch/pull/29068/files.